packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -617,11 +634,26 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
 
   private handleStateChange = (changes: StateChangeOptions<ShorthandValue>) => {
     if (changes.isOpen !== undefined && changes.isOpen !== this.state.open) {
-      this.trySetStateAndInvokeHandler('onOpenChange', null, { open: changes.isOpen })
+      const newState: { open: boolean; highlightedIndex?: number } = { open: changes.isOpen }
```

Comment:
Now we have these methods that are much more readable 👍 
Can we simplify more?

```jsx
const highlightedIndex = changes.isOpen ? this.getHighlightedIndexOnArrowKeyOpen(changes) : this.getHighlightedIndexOnClose(changes)
const newState = { open: changes.isOpen, highlightedIndex }

if (changes.isOpen && !this.props.search) {
  this.listRef.current.focus()
}
```

Reply:
the only solution is to have `const newState = { open: changes.isOpen, ...{highlightedIndex} }` to make sure that the highlightedIndex is not passed as undefined. otherwise Downshift will complain that we are transforming a controlled prop to uncontrolled.